src/forms/create_ticket.ts

```diff
@@ -20,3 +31,199 @@ export async function newCreateTicketForm(call: AppCall): Promise<AppForm> {
+                return;
+
+            case ZDFieldTypes.Integer:
+            case ZDFieldTypes.Decimal:
+            case ZDFieldTypes.Subject:
+            case ZDFieldTypes.Text:
+            case ZDFieldTypes.MultiLine: {
+                const f = new TextField(name);
+                f.setLabel(label);
+                if (isRequired) {
+                    f.isRequired();
+                }
+                if (field.type === ZDFieldTypes.MultiLine) {
+                    f.isTextArea();
+                }
+                this.builder.addField(f.toAppField());
+                return;
+            }
+
+            case ZDFieldTypes.Checkbox: {
+                const f = new BoolField(name);
+                f.setLabel(label);
+                if (isRequired) {
+                    f.isRequired();
+                }
+                this.builder.addField(f.toAppField());
+                return;
+            }
+
+            case ZDFieldTypes.TicketType:
+            case ZDFieldTypes.Priority:
+            case ZDFieldTypes.Tagger:
+            case ZDFieldTypes.Muliselect: {
+                const options = this.isSystemField(field) ? field.system_field_options : field.custom_field_options;
+                const f = new StaticSelectField(name, makeOptions(options));
+                f.setLabel(label);
+                if (isRequired) {
+                    f.isRequired();
+                }
+                if (field.type === ZDFieldTypes.Muliselect) {
+                    f.isMuliselect();
+                }
+                this.builder.addField(f.toAppField());
+                return;
+            }
+            default:
+                console.log('field type not mapped to app field. type =', field.type);
+                break;
+            }
+        });
+    }
+
+    isSystemField(field: Users.Fields.UserField): boolean {
+        return (SystemFields.includes(String(field.type)) || field.system_field_options);
+    }
+
+    // getMappedName gets the mapped field name for a zendesk field.
+    // custom zendesk field names are prefixed so that it can be easily parsed
+    // when form is submitted
+    getMappedName(field: Users.Fields.UserField): string {
+        const strFieldType = String(field.type);
+        switch (true) {
+        case strFieldType in MappedZDNames:
+            return MappedZDNames[strFieldType];
+        case this.isSystemField(field):
+            return strFieldType;
+        default :
+        }
+        return this.getCustomFieldName(field);
+    }
+
+    getCustomFieldName(field: Users.Fields.UserField): string {
+        return CreateTicketFields.PrefixCustomField + `${field.type}_` + field.id;
+    }
+
+    // addFormsToSelectField maps zendesk forms to a Mattermost select field
+    addFormsSelectField(): void {
+        const field = new StaticSelectField(CreateTicketFields.NameFormsSelect, makeFormOptions(this.zdTicketForms));
```

Comment:
I'd rather that we have smth. like : `const field = new StaticSelectField(...) { label:Form; refresh:...; is_required:...}` -  less verbose and contained and remove the one line helper functions (we don't get any special benefit from using them as wrappers around the field setting - like extra validation, checks, etc.)

Reply:
I understand.  I suppose these "helpers" aren't necessary and I need to make a parallel ticket to add the types for each field, instead of implementing this helper.

Reply:
done.  I removed this class and just construct an AppField